docs/topic/terraform.md

```diff
@@ -13,4 +13,10 @@ file with variable definitions for that cluster.
 
 Workspaces are stored centrally in the `two-eye-two-see-org` GCP project, even
 when we use Terraform for projects running on AWS / Azure. You must have
-access to this project before you can use terraform for our infrastructure.
\ No newline at end of file
+access to this project before you can use terraform for our infrastructure.
+
+## Cloud provider abstractions
+
+We will use completely different terraform code for each cloud provider,
```

Comment:
Would love to know what you think of this, @consideRatio @damianavila @sgibson91 @GeorgianaElena

Reply:
This is similar to what I was thinking about over lunch. Each cloud provider we deploy to via terraform has it's own submodule, and we have a top-level variable like `cloud_provider` that we use in some logic to import the correct module when doing a terraform plan/apply. It means more code to maintain, but as of yet I don't know of a more elegant solution and terraform isn't a _truly_ cloud-agnostic tool (it's dependent on the provider.

Reply:
> Each cloud provider we deploy to via terraform has it's own submodule, and we have a top-level variable like cloud_provider that we use in some logic to import the correct module when doing a terraform plan/apply.

I tried this, and unfortunately you can't use a variable to specify which module to load in terraform :(

Reply:
Boo! :( In which case, I think this solution is good enough.

Reply:
I wonder if @andersy005 or @cmd-ntrf with significant experience with terraform could comment about this for us? Anderson and Félix, do you think it is a reasonable approach to have separate terraform code for deploying infrastructure to separate cloud providers?

Reply:
I think this is a sensible proposal. Since terraform is not agnostic from the cloud provider I think any attempt to make that happen will look-like weird and, most likely, difficult if not impossible to do it in a sane way. But, hey, I do not have a lot of experience with terraform and I might be missing something...